### PR TITLE
Enemy

### PR DESCRIPTION
**Pull request by gaebos**

### CODE REVIEW

File: src/video.asm [hunk 1]
```diff
@@ -1,7 +1,3 @@
-.include "nes.asm"
```
Comment (voidexp) [inline src/video.asm:1]: let's revert this back

File: src/enemy.asm [hunk 2]
```diff
@@ -0,0 +1,442 @@
+;
+; Animation settings
+;
+.code
```
Comment (voidexp) [inline src/enemy.asm]: why this goes into `.code`? seems like `.rodata` is a better place for this stuff

File: src/enemy.asm [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+ANIMATION_SPEED = 8
+
+OFFSET_PADDING = 1
+OFFSET_TILE_ID = 2
+OFFSET_ATTRIBUTE_SET = 3       
+
+
+.segment "RAM"
+; 6 enemies, 5 bytes each
+current_enemy_set:  .res 40 ; 2 byte enemy address, 2 byte address, 1 byte anim state
```
Comment (voidexp) [inline src/enemy.asm]: but 6x5 makes 30... :/

File: src/enemy.asm [hunk 4]
```diff
@@ -0,0 +1,442 @@
+;
```
Comment (voidexp) [inline src/enemy.asm]: When it comes to use of global variables such as temp_1 and temp_address, all this code becomes NMI-unsafe.
In interrupt can occur after any instruction and if the interrupt handler itself modifies anything in temp_-stuff, after the execution returns to the subroutine, the data will be in unknown state and everything will be screwed.

The code should be rewritten to be context independent and use only locals on stack. This should be discussed and a generic mechanism developed.

**Review by voidexp — CHANGES_REQUESTED** [review]

File: src/video.asm (hunk 1 above)
Comment (gaebos) [reply src/video.asm:1]: the whole file I guess?

File: src/enemy.asm (hunk 3 above)
Comment (gaebos) [reply src/enemy.asm]: I have an excellent explanation for that!

**Pull request merged**